### PR TITLE
Add option to specify multiple source in file (bfs, sssp, bc)

### PR DESCRIPTION
**Pull request by gurbinder533**

### CODE REVIEW

**Review by arthurp — APPROVED** [review]
LGTM.

File: lonestar/analytics/cpu/betweennesscentrality/betweenness_centrality_cli.cpp [hunk 1]
```diff
@@ -31,16 +31,23 @@ namespace cll = llvm::cl;
...
     cll::Positional, cll::desc("<input file>"), cll::Required);
 
-static cll::opt<std::string> sourcesToUse(
-    "sourcesToUse",
-    cll::desc("Whitespace separated list of sources in a file to use in BC"),
+static cll::opt<std::string> startNodesFile(
+    "startNodesFile",
+    cll::desc("File containing whitespace separated list of source "
+              "nodes for computing betweenness-centrality; "
+              "if set, -startNodes and -numberOfSources is ignored"));
```
Comment (roshandathathri) [inline lonestar/analytics/cpu/betweennesscentrality/betweenness_centrality_cli.cpp:38]: numberOfSources is not ignored.

File: lonestar/analytics/cpu/betweennesscentrality/betweenness_centrality_cli.cpp [hunk 2]
```diff
@@ -31,16 +31,23 @@ namespace cll = llvm::cl;
...
+              "-startNodesFile is used"),
     cll::init(""));
-
-static cll::opt<unsigned int> numOfSources(
-    "numOfSources",
-    cll::desc("Number of sources to compute BC on (default all)"),
+static cll::opt<unsigned int> numberOfSources(
+    "numberOfSources",
+    cll::desc("Number of sources to compute betweenness-centrality on (default "
+              "all); pick first numberOfSources from -startNodesFile or "
```
Comment (roshandathathri) [inline lonestar/analytics/cpu/betweennesscentrality/betweenness_centrality_cli.cpp:48]: We should NOT have a default of "all" sources. Users would just use the default which is bad.

File: lonestar/analytics/cpu/bfs/bfs_cli.cpp [hunk 3]
```diff
@@ -37,13 +37,31 @@ static const char* url = "breadth_first_search";
...
+
+static cll::opt<std::string> startNodesFile(
+    "startNodesFile",
+    cll::desc("File containing whitespace separated list of source "
+              "nodes for computing breadth-first search; "
+              "if set, -startNodes is ignored"));
+static cll::opt<std::string> startNodesString(
+    "startNodes",
+    cll::desc("String containing whitespace separated list of source nodes for "
+              "computing betweenness centrality (default value '0'); ignore if "
```
Comment (roshandathathri) [inline lonestar/analytics/cpu/bfs/bfs_cli.cpp:49]: Says bc instead of bfs

File: lonestar/analytics/cpu/sssp/sssp_cli.cpp [hunk 4]
```diff
@@ -35,9 +35,24 @@ const char* url = "single_source_shortest_path";
...
-    cll::init(0));
+static cll::opt<std::string> startNodesFile(
+    "startNodesFile",
+    cll::desc("File containing whitespace separated list of source "
+              "nodes for computing single-source-shortest path search; "
+              "if set, -startNodes is ignored"));
+static cll::opt<std::string> startNodesString(
+    "startNodes",
+    cll::desc("String containing whitespace separated list of source nodes for "
+              "computing betweenness centrality (default value '0'); ignore if "
```
Comment (roshandathathri) [inline lonestar/analytics/cpu/sssp/sssp_cli.cpp:46]: Says bc instead of sssp

**Pull request merged**